### PR TITLE
cmockery: fix -flat_namespace and add source license

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Holding off on merge as may cause merge conflict with #102541

### CODE REVIEW

File: Formula/cmockery.rb
Comment (cho-m) [inline Formula/cmockery.rb:8]: From tarball:
- src/cmockery.c
    ```
     * Copyright 2008 Google Inc.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     * 
     * http://www.apache.org/licenses/LICENSE-2.0
    ```
- COPYING
    ```
    Copyright (c) 2006, Google Inc.
    All rights reserved.
    
    Redistribution and use in source and binary forms, with or without
    modification, are permitted provided that the following conditions are
    met:
    
        * Redistributions of source code must retain the above copyright
    notice, this list of conditions and the following disclaimer.
        * Redistributions in binary form must reproduce the above
    copyright notice, this list of conditions and the following disclaimer
    in the documentation and/or other materials provided with the
    distribution.
        * Neither the name of Google Inc. nor the names of its
    contributors may be used to endorse or promote products derived from
    this software without specific prior written permission.
    
    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
    ```

Adding `Apache-2.0` avoids having separate audit exception and matches state of most code/upstream.

File: audit_exceptions/permitted_formula_license_mismatches.json
Comment (cho-m) [inline audit_exceptions/permitted_formula_license_mismatches.json:4]: GitHub and formulae seem to match now (`MIT`). Didn't see any issue with `brew audit --online ...`

**Review by SMillerDev — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by cho-m** [comment]
Might need another review from PLC as audit codeowner due to rebase.

Rebase result should be similar to last review, so may just trigger merge at end of this week if no-one else does so and no other comments.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
